npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/vulnerability_auditor.rb

```diff
@@ -90,6 +90,13 @@ def viable_audit_result?(audit_result, security_advisories)
           validation_result = validate_audit_result(audit_result, security_advisories)
           return true if validation_result == :viable
 
+          case validation_result
+          when :fix_unavailable, :dependency_still_vulnerable, :downgrades_dependencies
+            raise "No patched version available"
+          when :vulnerable_dependency_removed
+            raise "Unable to upgrade vulnerable dependency. Upgrade manually"
+          end
```

Comment:
It feels off to raise from a predicate method (yes/no question) like this. Perhaps a minor refactoring could help?